src/routes/router.js

```diff
@@ -1,27 +1,46 @@
 const express = require('express');
 const axios = require('axios');
 const router = express.Router();
+const { writeToLocalCache, readLoadCache, lastModifiedDate } = require('../fs');
 
 router.get('/api/trending-service', (req, res) => {
-    res.setHeader('Content-Type', 'application/json');
-    const endpoint = process.env.ENDPOINT;
-    axios({
-        method: 'GET',
-        url: endpoint,
-        headers: {
-          "Accept": "application/json",
-          "Content-Type": "application/json"
-        }
-      }).then(response => {
-        res.send(JSON.stringify(response.data));
-      }).catch(error => {
-        res.status(400).send(JSON.stringify(error));
-      });
+  res.setHeader('Content-Type', 'application/json');
+  const endpoint = process.env.ENDPOINT;
+  axios({
+    method: 'GET',
+    url: endpoint,
+    headers: {
+      Accept: 'application/json',
+      'Content-Type': 'application/json'
+    }
+  })
+    .then(async response => {
+      res.send(JSON.stringify(response.data));
+      const lastModifyDate = await lastModifiedDate('localCache.json');
+      const lastModifyMonth = lastModifyDate.getMonth() + 1;
+      const lastModifyDay = lastModifyDate.getDate();
+      const currentDate = new Date().getDate();
+      const currentMonth = new Date().getMonth() + 1;
+      if (
+        lastModifyDate == 'undefined' ||
+        lastModifyMonth != currentMonth ||
+        lastModifyDay != currentDate
```

Comment:
no need to compare the datetime like this, just formate datetime to string and compare, like
lastModifyDate.toLocaleDateString("en-US") != currentDate.toLocaleDateString("en-US")

Reply:
That's good shot! Thanks Al~